eclipse-theia/theia
# Introduce custom select component

#### What it does

Closes https://github.com/eclipse-theia/theia/issues/8153

Introduces a custom react component for rendering `select` elements. It mirrors mostly how a native html `select` element works and allows for some additional features like element descriptions, details, separators, etc.

Mostly get's used for preference enum rendering, but I've also replaced some other `select` elements throughout the app.

#### How to test

In general, test that the component works just like the VSCode select element for enum preferences, launch debug config selection, etc.

1. Open the preference widget and play around with the different enum preferences. Also try to open preferences which are near the bottom of the app and observe that the dropdown gets inverted.
2. The component should react as expected when clicking on other elements or scrolling
3. The debug toolbar also uses the component. Confirm that every option is always rendered on a single line, expanding the dropdown width as necessary.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
